controllers:
  robot_state_controller:
    type: robot_state_controller/RobotStateController
    publish_rate: 50
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50
  gimbal_controller:
    type: rm_gimbal_controllers/Controller
    time_compensation: 0.03
    publish_rate: 100
    yaw:
      joint: "yaw_joint"
      upper: 1e9
      lower: -1e9
      pid: { p: 8, i: 10, d: 0.4, i_clamp_max: 0.3, i_clamp_min: -0.3, antiwindup: true, publish_state: true }
    pitch:
      joint: "pitch_joint"
      upper: 0.31
      lower: -1.39
      pid: { p: 20, i: 200, d: 0.4, i_clamp_max: 0.8, i_clamp_min: -0.8, antiwindup: true, publish_state: true }
    bullet_solver:
      resistance_coff_qd_10: 0.1
      resistance_coff_qd_15: 0.4
      resistance_coff_qd_16: 0.01
      resistance_coff_qd_18: 0.55
      resistance_coff_qd_30: 3.0
      g: 9.81
      delay: 0.18
      dt: 0.001
      timeout: 0.001
      publish_rate: 50
    moving_average_filter:
      is_debug: true
      center_offset_z: 0.05
      pos_data_num: 10
      vel_data_num: 30
      center_data_num: 100
      gyro_data_num: 100